wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSessionBuilder.swift

```diff
@@ -0,0 +1,347 @@
+        )
+        let mlsConversationVerificationStatusUpdater = MLSConversationVerificationStatusUpdater(
+            updateMLSGroupVerificationStatus: updateMLSGroupVerificationStatus,
+            syncContext: coreDataStack.syncContext
+        )
+        let mlsService = mlsService ?? MLSService(
+            context: coreDataStack.syncContext,
+            coreCryptoProvider: coreCryptoProvider,
+            conversationEventProcessor: ConversationEventProcessor(context: coreDataStack.syncContext),
+            userDefaults: .standard,
+            syncStatus: applicationStatusDirectory.syncStatus,
+            userID: coreDataStack.account.userIdentifier
+        )
+        let observeMLSGroupVerificationStatusUseCase = observeMLSGroupVerificationStatus ?? ObserveMLSGroupVerificationStatusUseCase(
+            mlsService: mlsService,
+            updateMLSGroupVerificationStatusUseCase: updateMLSGroupVerificationStatus,
+            syncContext: coreDataStack.syncContext
+        )
+        let proteusToMLSMigrationCoordinator = proteusToMLSMigrationCoordinator ?? ProteusToMLSMigrationCoordinator(
+            context: coreDataStack.syncContext,
+            userID: userId
+        )
+        let useCaseFactory = useCaseFactory ?? UseCaseFactory(
+            context: coreDataStack.syncContext,
+            supportedProtocolService: SupportedProtocolsService(context: coreDataStack.syncContext),
+            oneOnOneResolver: OneOnOneResolver(migrator: OneOnOneMigrator(mlsService: mlsService))
+        )
+
+        // setup builder
+
+        withAnalytics(analytics)
+        withAppVersion(appVersion)
+        withAppLock(appLock)
+        withApplication(application)
+        withApplicationStatusDirectory(applicationStatusDirectory)
+        withContextStorage(contextStorage)
+        withCoreCryptoProvider(coreCryptoProvider)
+        withCoreDataStack(coreDataStack)
+        withCryptoboxMigrationManager(cryptoboxMigrationManager)
+        withE2EIActivationDateRepository(e2eiActivationDateRepository)
+        withEARService(earService)
+        withFlowManager(flowManager)
+        withLastE2EIUpdateDateRepository(lastE2EIdentityUpdateDateRepository)
+        withLastEventIDRepository(lastEventIDRepository)
+        withMediaManager(mediaManager)
+        withMLSConversationVerificationStatusUpdater(mlsConversationVerificationStatusUpdater)
+        withMLSService(mlsService)
+        withObserveMLSGroupVerificationStatusUseCase(observeMLSGroupVerificationStatusUseCase)
+        withProteusToMLSMigrationCoordinator(proteusToMLSMigrationCoordinator)
+        withSharedUserDefaults(sharedUserDefaults)
+        withTransportSession(transportSession)
+        withUpdateMLSGroupVerificationStatusUseCase(updateMLSGroupVerificationStatus)
+        withUseCaseFactory(useCaseFactory)
+        withUserID(userId)
+    }
+
+    mutating func withAnalytics(_ analytics: (any AnalyticsType)?) {
```

Comment:
I suggest to get rid of all these functions and make the properties public.

Reply:
Thanks I've changed it